WcaOnRails/app/views/persons/_details.html.erb

```diff
@@ -1,6 +1,13 @@
 <div>
   <div class="text-center">
-    <h2><%= @person.name %></h2>
+    <h2>
+      <%= @person.name %>
+      <% if current_user.person == @person %>
```

Comment:
I believe #797 aims to show the link for admins as well.
Also note that if a user is not signed in, then `current_user` is `nil` and calling `.person` would result in an error.
We do even have a handy method indicating whether a user is able to edit another one:
https://github.com/thewca/worldcubeassociation.org/blob/dbdffcdc7e34fe496ad015e65f8a2b337556196f/WcaOnRails/app/models/user.rb#L412-L414
So I'd suggest
```ruby
if @person.user && current_user&.can_edit_user?(@person.user)
```